examples/object_detection_3d/object_detection_3d/workflow.py

```diff
@@ -0,0 +1,122 @@
+    image_projection: List[float]  # row-major (4x4)
+    right: Optional[ImageAsset] = None
+
+    def __post_init__(self) -> None:
+        if len(self.velodyne_to_camera_transformation) != 16:
+            raise ValueError(
+                f"{type(self).__name__} must have valid 16 floats \
+                               in velodyne_to_camera_transformation \
+                               ({len(self.velodyne_to_camera_transformation)} provided)",
+            )
+        if len(self.image_projection) != 16:
+            raise ValueError(
+                f"{type(self).__name__} must have valid 16 floats \
+                               in image_projection ({len(self.image_projection)} provided)",
+            )
+        if len(self.image_projection) != 16:
+            raise ValueError(
+                f"{type(self).__name__} must have valid 16 floats \
+                               in image_projection ({len(self.image_projection)} provided)",
+            )
+
+
+@dataclass(frozen=True)
+class AnnotatedBoundingBox3D(LabeledBoundingBox3D):
+    """
+    #Values    Name      Description
+    ----------------------------------------------------------------------------
+    1    type         Describes the type of object: 'Car', 'Van', 'Truck',
+                        'Pedestrian', 'Person_sitting', 'Cyclist', 'Tram',
+                        'Misc' or 'DontCare'
+    1    truncated    Float from 0 (non-truncated) to 1 (truncated), where
+                        truncated refers to the object leaving image boundaries
+    1    occluded     Integer (0,1,2,3) indicating occlusion state:
+                        0 = fully visible, 1 = partly occluded
+                        2 = largely occluded, 3 = unknown
+    1    alpha        Observation angle of object, ranging [-pi..pi]
+    4    bbox         2D bounding box of object in the image (0-based index):
+                        contains left, top, right, bottom pixel coordinates
+    3    dimensions   3D object dimensions: height, width, length (in meters)
+    3    location     3D object location x,y,z in camera coordinates (in meters)
+    1    rotation_y   Rotation ry around Y-axis in camera coordinates [-pi..pi]
+    1    score        Only for results: Float, indicating confidence in
+                        detection, needed for p/r curves, higher is better.
+    """
+
+    truncated: float
+    occluded: int
+    alpha: float
+    difficulty: str
+
+
+@dataclass(frozen=True)
+class GroundTruth(BaseGroundTruth):
+    total: int
+    easy_image_bboxes: List[LabeledBoundingBox]
+    easy_velodyne_bboxes: List[AnnotatedBoundingBox3D]
+    moderate_image_bboxes: List[LabeledBoundingBox]
+    moderate_velodyne_bboxes: List[AnnotatedBoundingBox3D]
+    hard_image_bboxes: List[LabeledBoundingBox]
+    hard_velodyne_bboxes: List[AnnotatedBoundingBox3D]
+    other_image_bboxes: List[LabeledBoundingBox]
+    other_velodyne_bboxes: List[AnnotatedBoundingBox3D]
```

Comment:
What is an "other" bbox?

Reply:
This is something I need to figure out if my understanding of difficulty criteria is correct or not, if current separation of groundtruth bboxes by difficulty is desired over original implementation of single list.

Reply:
I'm interpreting conditions specified in kitti 3d eval page as conjunctive. With that implementation, there are objects that do not fall into 3 defined categories.

Cross referencing [mmdetection3d's implenetation](https://github.com/open-mmlab/mmdetection3d/blob/main/tools/dataset_converters/kitti_data_utils.py#L621), which also has a catch clause for that.